api/v1alpha1/tinkerbell_type.go

```diff
@@ -0,0 +1,125 @@
+// CRs is Tinkerbell which the operator will install the tink-stack based on its specs. Once the CR is deleted,
+// the operator will delete all tinkerbell resources.
+type Tinkerbell struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec describes the desired tinkerbell stack state.
+	Spec TinkerbellSpec `json:"spec,omitempty"`
+}
+
+// TinkerbellSpec specifies details of an Tinkerbell setup.
+type TinkerbellSpec struct {
+	// TinkerbellVersion is the tinkerbell crd version.
+	TinkerbellVersion string `json:"tinkerbellVersion"`
+
+	// ClusterDNS is the ip address of the cluster dns resolver.
+	ClusterDNS string `json:"clusterDNS"`
+
+	// OverwriteRegistry is the registry to use for all images. If this field is set, all tink service deployment images
+	// will be prefixed with this value. For example if the value here was set to docker.io, then boots image will be
+	// docker.io/tinkerbell/boots.
+	OverwriteRegistry string `json:"overwriteRegistry"`
+
+	// DockerPullConfig the secret name containing the docker auth config which should exist in the same namespace where
+	// the operator is deployed(typically tinkerbell)
+	DockerPullComfig string `json:"dockerPullComfig"`
+
+	// Boots contains all the information and spec about boots.
+	Boots BootsSpec `json:"boots"`
+	// Hegel contains all the information and spec about boots.
+	Hegel HegelSpec `json:"hegel"`
+	// Rufio contains all the information and spec about rufio.
+	Rufio RufioSpec `json:"rufio"`
+	// Tink contains all the information and spec about tink.
+	Tink TinkSpec `json:"tink"`
+}
+
+// ImageSpec specifies the details of a tinkerbell services images.
+type ImageSpec struct {
+	// ImageRepository is used to set the BootsSpec image repository.
+	ImageRepository string `json:"imageRepository,omitempty"`
+
+	// ImageTag is used to set the BootsSpec image tag.
+	ImageTag string `json:"imageTag,omitempty"`
+}
+
+// BootsSpec specifies the details of tinkerbell service boots.
+type BootsSpec struct {
+	// Image specifies the details of a tinkerbell services images
+	Image ImageSpec `json:"image"`
+
+	// DHCPAddress set the ip and port to listen on for DHCP.
+	DHCPAddress string `json:"dhcpAddress"`
+
+	// TrustedProxies comma separated allowed CIDRs subnets to be used as trusted proxies
+	TrustedProxies string `json:"trustedProxies"`
+
+	// FacilityCode represents the facility in use.
+	FacilityCode string `json:"facilityCode"`
+
+	// HTTPBind is the port to listen on for the serving iPXE binaries and files via HTTP.
+	HTTPBind int `json:"httpBind"`
+
+	// MirrorBaseURL the URL from where the "OSIE" or Hook kernel(s) and initrd(s) will be downloaded by netboot clients
+	MirrorBaseURL string `json:"mirrorBaseURL"`
+
+	// OSIEPathOverride override the URL where OSIE/Hook images are located
+	OSIEPathOverride string `json:"osiePathOverride"`
```

Comment:
Boots top of tree and version `v0.9.0` dont have this value any more. i would suggest we use `v0.9.0` values if possible.

Reply:
The thing is, these fields are inspired by the charts, not by the projects directly. In other words, I copied all these params out of the boots template in the charts repo.

Reply:
Which chart version? I don't see most of these values in the chart top of tree/latest version.

Reply:
I am looking into the boots chart and it says v.0.8.0 https://github.com/tinkerbell/charts/blob/main/tinkerbell/boots/values.yaml#L8-L9 I will open a PR to bump the boots version and adjust the params